--- awstats.pl.orig	2013-07-04 23:07:34.402662237 +0200
+++ awstats.pl	2013-07-04 23:07:41.530556705 +0200
@@ -1706,29 +1706,10 @@
 	# Other possible directories :				"/usr/local/etc/awstats", "/etc"
 	# FHS standard, Suse package : 				"/etc/opt/awstats"
 	my $configdir         = shift;
-	my @PossibleConfigDir = (
-			"$DIR",
-			"/etc/awstats",
-			"/usr/local/etc/awstats", "/etc",
-			"/etc/opt/awstats"
-		); 
 
-	if ($configdir) {
-		# Check if configdir is outside default values.
-		my $outsidedefaultvalue=1;
-		foreach (@PossibleConfigDir) {
-			if ($_ eq $configdir) { $outsidedefaultvalue=0; last; }
-		}
-
-		# If from CGI, overwriting of configdir with a value that differs from a defautl value
-		# is only possible if AWSTATS_ENABLE_CONFIG_DIR defined
-		if ($ENV{'GATEWAY_INTERFACE'} && $outsidedefaultvalue && ! $ENV{"AWSTATS_ENABLE_CONFIG_DIR"})
-		{
-			error("Sorry, to allow overwriting of configdir parameter, from an AWStats CGI page, with a non default value, environment variable AWSTATS_ENABLE_CONFIG_DIR must be set to 1. For example, by adding the line 'SetEnv AWSTATS_ENABLE_CONFIG_DIR 1' in your Apache config file or into a .htaccess file.");
-		}
-
-		@PossibleConfigDir = ("$configdir");
-	}
+	error("Wrong config name") unless( $SiteConfig =~ /^[0-9a-z]+$/ );
+	my $vhffsconfigdir = '/data/logs/repository/parsed/'.$SiteConfig.'/awstats';
+	my @PossibleConfigDir=("$vhffsconfigdir");
 
 	# Open config file
 	$FileConfig = $FileSuffix = '';
@@ -17363,7 +17344,7 @@
 else { @DOWIndex = ( 0, 1, 2, 3, 4, 5, 6 ); }
 
 # Should we link to ourselves or to a wrapper script
-$AWScript = ( $WrapperScript ? "$WrapperScript" : "$DirCgi$PROG.$Extension" );
+$AWScript = ( $WrapperScript ? "$WrapperScript" : "/$DirCgi$PROG.$Extension" );
 if (index($AWScript,'?')>-1) 
 {
     $AWScript .= '&amp;';   # $AWScript contains URL parameters
